src/parachain/amm.ts

```diff
@@ -299,32 +321,126 @@ export class DefaultAMMAPI implements AMMAPI {
         return decodeNumberOrHex(rawA);
     }
 
-    private async _getStableLiquidityPool(
+    private _getStableBasePooledCurrenciesAdjustedToLpTokenAmount(
+        basePooledCurrencies: PooledCurrencies,
+        lpTokenTotalSupply: MonetaryAmount<StableLpToken>,
+        metaPoolLpTokenAmount: MonetaryAmount<StableLpToken>
+    ) {
+        const changeCoefficient = metaPoolLpTokenAmount.div(lpTokenTotalSupply.toBig()).toBig();
+        return basePooledCurrencies.map((amount) => amount.mul(changeCoefficient));
+    }
+
+    private async _getStableLiquidityPoolData(
         poolId: number,
-        poolData: ZenlinkStableAmmPrimitivesPool
-    ): Promise<StableLiquidityPool | null> {
-        const poolBase = DefaultAMMAPI.getStableBasePool(poolData);
-        if (poolBase === null) {
+        poolData: ZenlinkStableAmmPrimitivesPool,
+        metaPoolLpTokenAmount?: MonetaryAmount<StableLpToken>
+    ) {
+        const poolInfo = DefaultAMMAPI.getStablePoolInfo(poolData);
+        if (poolInfo === null) {
             return null;
         }
 
         const [pooledCurrencyIds, pooledCurrencyBalances, tradingFee] = [
-            poolBase.currencyIds,
-            poolBase.balances,
+            poolInfo.currencyIds,
+            poolInfo.balances,
             // TODO: check number base for fee
-            decodeFixedPointType(poolBase.fee),
+            decodeFixedPointType(poolInfo.fee),
         ];
-
-        const lpToken = DefaultAMMAPI.getStableLpTokenFromPoolData(poolId, poolBase);
-
-        const [pooledCurrencies, apr, A, totalSupply] = await Promise.all([
+        const lpToken = DefaultAMMAPI.getStableLpTokenFromPoolData(poolId, poolInfo);
+        const [pooledCurrenciesBase, apr, A, totalSupply] = await Promise.all([
```

Comment:
Maybe a good time to rename `A` to `amplificationCoefficient` or `amplCoeff` or something... not `A` :P

Reply:
good idea !